libs/core/src/lib/select/select-selection-model.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { SelectionModel } from '@angular/cdk/collections';
+
+import { OptionComponent } from './option/option.component';
+
+@Injectable()
+export class SelectSelectionModelService {
+    _selectionModel: SelectionModel<OptionComponent>;
+
+    /**
+     * Returns true if select has any value selected
+     */
+    get empty(): boolean {
+        return !this._selectionModel || this._selectionModel.isEmpty();
+    }
+
+    /**
+    * Returns true if select has any value selected
+    */
+    clear(): void {
+        this._selectionModel.clear();
+    }
+
+}
```

Comment:
What is the benefit of moving selection model in this service?
This service does nothing and exposes `SelectionModel` anyway.

Reply:
keyManager and selection models are provided as service with commonly used functions.
By targeting future  enhancements if any and for other components for reusability.

Reply:
In my understanding the main purpose of this service should be incapsulation.
If we want to incapsulate `SelectionModel` we need to hide direct reference and define public api (actually I don't see much sense in it cause `SelectionModel` already incapsulated well by itself)

In case if you want to extract some selection component logic I don't see any in this service.
As for me it's useless.

By the way `selection.clear` method is not used at all. All manipulations with selectionModel goes directly through reference to the `_selectionModel` itself.

Reply:
With respect to selectionmodel I agree with you, 

 @JKMarkowski could you please share your thoughts on it, I can see Dimitry also sharing same concern as I did earlier.

Reply:
Hi @JKMarkowski , as discussed we won't be having separate service for selectioonmodel